File: app/schemas/bus.py
from pydantic import BaseModel, field_validator

class BusRoutes(BaseModel):
    route_id: int
    route_short_name: str
    route_long_name: str

class BusStops(BaseModel):
    stop_id: int
    stop_name: str
    stop_lat: float
    stop_lon: float

class RouteStops(BaseModel):
    route_stop_id: int
    route_id: int
    stop_id: int
    stop_sequence: int

File: frontend/src/types/bus.ts
export interface Route {
  route_id: number;
  route_short_name: string;
  route_long_name: string;
}

export interface Stop {
  route_stop_id: number;
  stop_id: number;
  stop_name: string;
  stop_lat: number;
  stop_lon: number;
  stop_sequence: number;
}

export interface LivePosition {
  route_name: string;
  latitude: number;
  longitude: number;
  velocity: number;
  stop_name: string;
  tempo_chegada: string;
}
